### DISCUSSION TITLE
Error in deleting Sample-Apps

### DISCUSSION THREAD

**Discussion by Harigovind-AU** [gruntwork-io/knowledge-base#658] [Help]
### PROBLEM STATEMENT
I am trying to delete sample apps which are running inside AWS EKS and I followed the gruntwork's document (docs provided with reference architecture) for deleting sample-apps. We are running our pipeline in Gitlab and in plan stage on a feature branch, deleting sample apps is showing desired outputs and pipeline is getting succeeded but later in plan stage it is throwing the below error:

![MicrosoftTeams-image (3)](https://user-images.githubusercontent.com/114720109/217155542-48c0e8a6-d98a-430d-b039-eb630886628d.png)

### QUESTIONS

- Is there any protection enabled that stops from deleting something in currently deployed infrastructure or stops from un-deploying sample apps on main branch?
- If there is no such protection, mentioned in above question, what are we missing in deleting sample apps?

---

<ins datetime="2023-02-06T05:09:48Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109869">Tracked in ticket #109869</a></p>
</ins>

**Comment by pras111gw** [comment]
Hi @Harigovind-AU, Thank you for reaching out to us. Can you attach output of the job that was run to deploy these sample services? The logs indicate no .TF files in current folder. Please attach output of `ls -al` command in current and parent folder as well.

**Reply by hemanth-chinnammagari** [reply]
Hi @pras111gw, thanks for your reply. The outputs of `ls -al` only contain the README.md and terragrunt.hcl file in current folder as we are trying to undeploy sample apps using pipeline so there won't be any .tf files locally in current or parent folder. Also plan is getting successful in some other branch and error is only coming in main branch.
The sample apps were deployed by gruntwork and we got the reference architecture with these sample apps deployed, so currently we don't have the output of the job that was run to deploy these sample services.

**Reply by pras111gw** [reply]
Hi @hemanthc1234 thanks for providing additional context. Are you using one of the examples from here https://github.com/gruntwork-io/terraform-aws-service-catalog/tree/master/examples/for-learning-and-testing or something else?

About your questions - in general if there are protected resources (delete protection on RDS - for example), `destroy` command would fail. That does not look like the case here.

`destroy` command needs two things, a reference to state (should be in terragrunt.hcl), and the resources to destroy. Can you paste the `.hcl` file so we can trace it back to the path for resources?
